pkg/util/errors.go

```diff
@@ -100,6 +100,37 @@ func (es MultiError) IsDeadlineExceeded() bool {
 	return true
 }
 
+// GroupedErrors implements the error interface, and it contains the errors used to construct it
+// grouped by the error message.
+type GroupedErrors struct {
+	MultiError
+}
+
+// Error Returns a concatenated string of the errors grouped by the error message along with the number of occurrences
+// of each error message.
+func (es GroupedErrors) Error() string {
+	mapErrs := make(map[string]int, len(es.MultiError))
+	for _, err := range es.MultiError {
```

Comment:
> Would it make more sense to just store the map, like you have here, with counts by error message?

I did this to inherit all but the Error() method from `MultiLine`. IMO having to reimplement the rest of methods is not justified by the (likely?) slight speed-up in the Errors() method.

> I haven't looked at MultiError, but it appears to just be storing all the errors. Would it make more sense to just store the map, like you have here, with counts by error message? That may alleviate some of your concerns about ballooning the size of the errors too much?

Can you elaborate on how it would help to aliviate that?

Reply:
talked about off line, and due to the slight difference in error messages we'd need a better mechanism to group them in order for this to work, so I agree on not doing this now.